tests/components/recorder/test_util.py

```diff
@@ -107,6 +107,7 @@ async def test_last_run_was_recently_clean(hass):
         is False
     )
 
+    # We should let the recorder thread do this instead, not call directly
```

Comment:
Comment added to explain why the test fails, to be removed before the PR is merged

Reply:
It might be useful to run a few of these with an actual on disk db (just 3-4 to keep the overhead low) to make sure we test that path as well.